compile error on gcc16

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/fory/issues) and found no similar issues.


### Version

```shell
[ 22%] Building CXX object _deps/fory-build/fory/util/CMakeFiles/fory_util.dir/error.cc.o
包含于文件 /home/xffish/workspace_cpp/use-cmake/build/_deps/fory-src/cpp/fory/util/error.cc:20:
/home/xffish/workspace_cpp/use-cmake/build/_deps/fory-src/cpp/fory/util/error.h:107:30: 错误：‘uint32_t’未声明
  107 |   static Error type_mismatch(uint32_t type_a, uint32_t type_b) {
      |                              ^~~~~~~~
/home/xffish/workspace_cpp/use-cmake/build/_deps/fory-src/cpp/fory/util/error.h:26:1: 附注：‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   25 | #include <utility>
  +++ |+#include <cstdint>
   26 | 
/home/xffish/workspace_cpp/use-cmake/build/_deps/fory-src/cpp/fory/util/error.h:107:47: 错误：‘uint32_t’未声明
  107 |   static Error type_mismatch(uint32_t type_a, uint32_t type_b) {
      |                                               ^~~~~~~~
/home/xffish/workspace_cpp/use-cmake/build/_deps/fory-src/cpp/fory/util/error.h:107:47: 附注：‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
make[2]: *** [_deps/fory-build/fory/util/CMakeFiles/fory_util.dir/build.make:93：_deps/fory-build/fory/util/CMakeFiles/fory_util.dir/error.cc.o] 错误 1
make[1]: *** [CMakeFiles/Makefile2:6104：_deps/fory-build/fory/util/CMakeFiles/fory_util.dir/all] 错误 2
make: *** [Makefile:136：all] 错误 2
```

gcc (GCC) 16.0.1 20260118 (experimental)

`cmake -B build && cmake --build build` can reproduce.

[use-cmake.zip](https://github.com/user-attachments/files/24863956/use-cmake.zip)


### Component(s)

C++

### Minimal reproduce step

use attachment use-cmake.zip blow.

`cmake -B build && cmake --build build -j`



### What did you expect to see?

compile successful

### What did you see instead?

nop

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

--- a/cpp/fory/util/error.h
+++ b/cpp/fory/util/error.h
@@ -19,6 +19,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <memory>
 #include <ostream>
 #include <string>

